xonsh/tokenize.py

```diff
@@ -330,10 +330,17 @@ def maybe(*choices):
     "1>2",
 )
 IORedirect = group(group(*_redir_map), "{}>>?".format(group(*_redir_names)))
-_redir_check = set(_redir_map)
-_redir_check = {"{}>".format(i) for i in _redir_names}.union(_redir_check)
-_redir_check = {"{}>>".format(i) for i in _redir_names}.union(_redir_check)
-_redir_check = frozenset(_redir_check)
+
+
+def _get_redir_check():
+    _map = set(_redir_map)
+    _check1 = {"{}>".format(i) for i in _redir_names}.union(_map)
+    _check2 = {"{}>>".format(i) for i in _redir_names}.union(_check1)
+    return frozenset(_check2)
```

Comment:
Why is this function needed?

Reply:
to replace the imperative black that  used the same variable name. Also this is very outdated code taken from older python version. the mypy was complaining about it.

Reply:
Is there a way to keep it imperative by changing variable names?

Reply:
yes we can do it. Will it also end up in global namespace ?

Reply:
Yes, unless deleted, because tokenize is not part of the [excluded list](https://github.com/xonsh/xonsh/blob/b5a7439dca1ba9f4d587889daea2b90e1b128d49/xonsh/__init__.py#L4-L6)

Reply:
so can we replace this with the latest one from the Python's stdlib ?

Reply:
I have replaced the func with statements.